app/scripts/metamask-controller.js

```diff
@@ -4616,7 +4627,11 @@ export default class MetamaskController extends EventEmitter {
     try {
       // Scan accounts until we find an empty one
       const chainId = this.#getGlobalChainId();
+      let entropySource = keyringId;
 
+      if (!entropySource) {
+        entropySource = this.keyringController.state.keyringsMetadata[0].id;
+      }
```

Comment:
will this be unused when the fenced code is removed

Reply:
good catch, thanks!